src/internal/coredb/projects_test.go

```diff
@@ -0,0 +1,153 @@
+	require.NoError(t, err, "should be able to create a tx")
+	defer tx.Rollback()
+	migrationEnv := migrations.Env{EtcdClient: testetcd.NewEnv(ctx, t).EtcdClient}
+	require.NoError(t, migrations.ApplyMigrations(ctx, db, migrationEnv, clusterstate.DesiredClusterState), "should be able to set up tables")
+	createInfo := &pfs.ProjectInfo{Project: &pfs.Project{Name: testProj}, Description: testProjDesc, CreatedAt: types.TimestampNow()}
+	require.NoError(t, CreateProject(ctx, tx, createInfo), "should be able to create project")
+	// the 'default' project is ID 1.
+	getInfo, err := GetProject(ctx, tx, 2)
+	fmt.Printf("%q %q\n", createInfo, getInfo)
+	require.NoError(t, err, "should be able to get a project")
+	require.NoError(t, tx.Commit())
+	require.Equal(t, createInfo.Project.Name, getInfo.Project.Name)
+	require.Equal(t, createInfo.Description, getInfo.Description)
+}
+
+func TestListProject(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	db := dockertestenv.NewTestDB(t)
+	tx, err := db.BeginTxx(ctx, nil)
+	require.NoError(t, err, "should be able to create a tx")
+	defer tx.Rollback()
+	migrationEnv := migrations.Env{EtcdClient: testetcd.NewEnv(ctx, t).EtcdClient}
+	require.NoError(t, migrations.ApplyMigrations(ctx, db, migrationEnv, clusterstate.DesiredClusterState), "should be able to set up tables")
+	require.NoError(t, DeleteProject(ctx, tx, "default"))
+	size := 210
+	expectedInfos := make([]*pfs.ProjectInfo, size)
+	for i := 0; i < size; i++ {
+		createInfo := &pfs.ProjectInfo{Project: &pfs.Project{Name: fmt.Sprintf("%s%d", testProj, i)}, Description: testProjDesc, CreatedAt: types.TimestampNow()}
+		expectedInfos[i] = createInfo
+		require.NoError(t, CreateProject(ctx, tx, createInfo), "should be able to create project")
+	}
+	iter, err := ListProject(ctx, tx)
+	require.NoError(t, err, "should be able to list projects")
+	i := 0
+	require.NoError(t, stream.ForEach[*pfs.ProjectInfo](ctx, iter, func(proj *pfs.ProjectInfo) error {
+		if err != nil {
+			require.NoError(t, err, "should be able to iterate over projects")
+		}
+		require.Equal(t, expectedInfos[i].Project.Name, proj.Project.Name)
+		require.Equal(t, expectedInfos[i].Description, proj.Description)
+		i++
+		return nil
+	}))
+}
+
+func TestDeleteAllProjects(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	db := dockertestenv.NewTestDB(t)
+	tx, err := db.BeginTxx(ctx, nil)
+	require.NoError(t, err, "should be able to create a tx")
+	defer tx.Rollback()
+	migrationEnv := migrations.Env{EtcdClient: testetcd.NewEnv(ctx, t).EtcdClient}
+	require.NoError(t, migrations.ApplyMigrations(ctx, db, migrationEnv, clusterstate.DesiredClusterState), "should be able to set up tables")
+	size := 3
+	for i := 0; i < size; i++ {
+		createInfo := &pfs.ProjectInfo{Project: &pfs.Project{Name: fmt.Sprintf("%s%d", testProj, i)}, Description: testProjDesc, CreatedAt: types.TimestampNow()}
+		require.NoError(t, CreateProject(ctx, tx, createInfo), "should be able to create project")
```

Comment:
Putting `i` in the message is always helpful.  Does the first iteration fail, or the last?